multiversx/mx-chain-go
# Link inner tx to results

## Reasoning behind the pull request
- missing link between tx results and inner txs
  
## Proposed changes
- linked results to inner txs
- now GetTransaction(innerTxHash) should return the inner tx as well with the results generated

## Testing procedure
- standard system test + full relayed v3 testing with lots of scenarios + relayed v1/v2 tests that confirm no change happened in terms of generated scrs/logs

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?
